app/assets/javascripts/ss/lib/image_editor.js

```diff
@@ -0,0 +1,133 @@
+SS_ImageEditor = function (el) {
+  var pThis = this;
+
+  this.$el = $(el);
+  this.cropper = new Cropper(
+    this.$el.find('img.target')[0],
+    {
+      zoomOnWheel: false,
+      ready: function(e) {
+        pThis.updateInspect();
+      },
+      cropmove: function(e) {
+        pThis.updateInspect();
+      }
+    }
+  );
+
+  this.$el.find('.toolbar button').on('click', function (e) {
+    var func = $(this).data('func');
+    if (func && pThis[func]) {
+      pThis[func]();
+
+      e.preventDefault();
+      return false;
+    }
+  });
+
+  this.$el.find('.toolbar select[name=size]').on('change', function () {
+    var val = $(this).val();
+    if (! val) {
+      return;
+    }
+
+    var sizeArray = val.split(',');
+    pThis.changeSize(parseInt(sizeArray[0], 10), parseInt(sizeArray[1], 10));
+  });
+};
+
+SS_ImageEditor.prototype = {
+  inspect: function () {
+    if (!this.cropper) {
+      return;
+    }
+
+    var data = this.cropper.getData(true);
+    console.log(data);
+    var imageData = this.cropper.getImageData();
+    console.log(imageData);
+    var canvasData = this.cropper.getCanvasData();
+    console.log(canvasData);
+  },
+
+  updateInspect: function() {
+    var data = this.cropper.getData(true);
+    this.$el.find('.toolbar input[name=x]').val(data.x);
+    this.$el.find('.toolbar input[name=y]').val(data.y);
+    this.$el.find('.toolbar input[name=width]').val(data.width);
+    this.$el.find('.toolbar input[name=height]').val(data.height);
+  },
+
+  changeSize: function(width, height) {
+    var data = this.cropper.getData(false);
+
+    data.width = width;
+    data.height = height;
+
+    var data = this.cropper.setData(data);
```

Comment:
__[jshint]__
* 'data' is already defined.

<!-- Comment by [SideCI](https://sideci.com) -->